xla/backends/gpu/runtime/nvshmem_put_thunk.cc

```diff
@@ -0,0 +1,178 @@
+/* Copyright 2025 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "xla/backends/gpu/runtime/nvshmem_put_thunk.h"
+
+#include <cstdint>
+#include <optional>
+#include <utility>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "xla/backends/gpu/collectives/gpu_collectives.h"
+#include "xla/service/collective_ops_utils.h"
+#include "xla/backends/gpu/runtime/nvshmem_collective_thunk.h"
+#include "xla/backends/gpu/runtime/thunk.h"
+#include "xla/core/collectives/communicator.h"
+#include "xla/hlo/ir/hlo_instruction.h"
+#include "xla/hlo/ir/hlo_instructions.h"
+#include "xla/service/collective_ops_utils.h"
+#include "xla/service/gpu/backend_configs.pb.h"
+#include "xla/service/gpu/transforms/collectives/collective_ops_utils.h"
+#include "xla/status_macros.h"
+#include "xla/stream_executor/stream.h"
+#include "xla/primitive_util.h"
+#include "tsl/platform/errors.h"
+#include "tsl/platform/logging.h"
+#include "tsl/platform/statusor.h"
+
+namespace xla {
+namespace gpu {
+
+NvshmemPutThunk::NvshmemPutThunk(ThunkInfo thunk_info,
+                                 const HloSendInstruction* instr,
+                                 int64_t replica_count, int64_t partition_count,
+                                 const CollectiveThunk::Buffer& buffer)
+    : NvshmemCollectiveThunk(Thunk::kNvshmemPut, thunk_info,
+                             IsGPUSyncCollective(*instr)),
+      config_(GetNvshmemP2PConfigForPutGet(instr, instr->operand(0)->shape(),
+                                           replica_count, partition_count)),
+      buffer_(buffer),
+      execution_counters_(config_.validation_kind ==
+                                  NvshmemP2PConfig::ValidationKind::kConditional
+                              ? new NvshmemP2PExecutionCounters()
```

Comment:
```suggestion
                              ? std::make_shared<NvshmemP2PExecutionCounters>()
```
Or `std::make_unique` if stored as `std::unique_ptr`